unlock-app/src/hooks/useGetReferrers.ts

```diff
@@ -0,0 +1,17 @@
+import { useQuery } from '@tanstack/react-query'
+import { PaywallConfigType } from '@unlock-protocol/core'
+
+import { getReferrers } from '~/utils/checkoutLockUtils'
+
+export const useGetReferrers = (
```

Comment:
I don't understand why we need a hook for this and we could not call `getReferrers` when we need ?

Reply:
(you actually do that in several places and that's correct!)

Reply:
since `getReferrers` is not sync and it does make rpc calls, I am using hook. react query  caches it based on the inputs to avoid making extra calls on re-renders causing issues.